internal/compiler/generator/cpp.rs

```diff
@@ -1442,12 +1441,10 @@ fn generate_item_tree(
         }),
     ));
 
-    let mut destructor = vec!["auto self = this;".to_owned()];
-
-    destructor.push(format!(
-        "if (auto &window = {}->m_window) window->window_handle().unregister_component(self, item_array());",
-        root_access
-    ));
+    let root_access = if parent_ctx.is_some() { "root" } else { "this" };
```

Comment:
I would've thought that we could use this everywhere perhaps.

Reply:
only the root has a direct pointer to the window